xgcm/test/test_grid.py

```diff
@@ -783,13 +795,17 @@ def test_keep_coords(funcname, gridtype):
         result = func(ds.tracer, axis_name)
         base_coords = list(result.dims)
         augmented_coords = [
-            c for c in ds.tracer.coords if set(ds[c].dims).issubset(result.dims)
+            c
+            for c in ds.coords
+            if set(ds[c].dims).issubset(result.dims) and c not in result.dims
         ]
+
         if funcname in ["integrate", "average"]:
             assert set(result.coords) == set(base_coords + augmented_coords)
         else:
             assert set(result.coords) == set(base_coords)
 
+        # TODO: why is the behavior different for integrate and average?
```

Comment:
What is different about them?

Reply:
I think this is ultimately related to a situation as #458.

Reply:
Thinking again about this. Maybe I was not hundred percent right. Its a similar situation (as in this does not go through the ufunc logic), but why that ultimately does not preserve the coordinates is a bit weird. Probably for a future PR too though.